server/routes/api/browser.py

```diff
@@ -164,10 +163,27 @@ def get_statvar_hierarchy(dcid):
     for svg_id, svg in svg_map.items():
         if svg_id in seen_sv:
             continue
+        svg["level"] = 0
         processed_svg_map[svg_id] = svg
         statvar_hierarchy_helper(svg_id, svg_map, processed_svg_map,
-                                 processed_sv, seen_sv)
+                                 processed_sv, seen_sv, 1)
+    for sv_id, sv in processed_sv.items():
+        parent_svg = processed_svg_map.get(sv["parent"])
+        sv["level"] = parent_svg["level"] + 1
     result = {}
     result["statVarGroups"] = processed_svg_map
     result["statVars"] = processed_sv
     return Response(json.dumps(result), 200, mimetype='application/json')
+
+
+@bp.route('/search_statvar_hierarchy')
+@cache.cached(timeout=3600 * 24, query_string=True)
+def search_statvar_hierarchy():
+    """Gets the statvars and statvar groups that match the tokens in the query
+    """
+    query = request.args.get("query").lower()
+    query = query.replace(",", " ")
+    tokens = query.split()
+    token_string = "^".join(tokens)
```

Comment:
can directly pass in tokens instead "^" joined tokens?

Reply:
done